package-res/resources/web/vizapi/VizController.js

```diff
@@ -716,29 +716,31 @@ pentaho.VizController.getRgbGradientFromMultiColorHex = function(value, min, max
   var steps = colors.length-1;
   var range = max-min;
 
+  var startStep;
+  var endStep;
   if(range <= 0) {
-    var start = colors.length-1;
-    var end = start;
+    startStep = colors.length-1;
+    endStep = startStep;
   } else {
-    var start = Math.floor(((value-min)/range) * steps);
-    var end = Math.ceil(((value-min)/range) * steps);
+    var inRange = ((value-min)/range);
+    startStep = Math.floor(inRange * steps);
+    endStep = Math.ceil(inRange * steps);
   }
```

Comment:
Are you sure this is right? The difference between  `startStep` and `endStep` is either 0 or 1, which means `color1` and `color2` will either be the same color or adjacent.